### DISCUSSION TITLE
Finish ToDos

### DISCUSSION THREAD

**Discussion by Elwood4415** [702573N/Obsidian-Tasks-Calendar#20] [General]
First, I appreciate your work on this. I have just installed it, but it already looks quite interesting. I do, however, have a question. Is there any way to mark a task complete using tasksCalendar? I am guessing that this feature is not available, but want to make sure that I am not missing something. I do realize that by clicking on a task the page on which the task is located is opened. And that is helpful. Thanks again!

**Comment by 702573N** [comment]
> First, I appreciate your work on this.

_Thank you for these kind words 😀_

Unfortunately, I don't see any possibility of implementing this at the moment. The main problem here is that the Tasks calendar was completely implemented with the [Dataview](https://github.com/blacksmithgu/obsidian-dataview) plugin. However, the Dataview plugin handles tasks differently than the [Tasks](https://github.com/obsidian-tasks-group/obsidian-tasks) plugin and has much fewer options here. An example here are the recurring tasks. Such tasks are not provided for in Dataview at all and therefore cannot be processed correctly. With the Tasks plugin, a recurring task is recognised, the task is completed and the completion date is appended, and a new task with a new to-do date is created. Eventually, the Tasks calendar should become part of the Tasks plug-in. I had already spoken to @claremacrae about this, but the dataview problem also comes into play here. The code implemented in Dataview cannot simply be integrated into a plug-in, but would have to be almost completely rewritten. This is a lot of work, because calendars are unfortunately very complex and have their pitfalls.

**Reply by claremacrae** [reply]
@702573N is of course exactly right here.

It's possible that myself and another plugin developer may be able to expose the Tasks recurrence code for use in other plugins.
In which case, it might be possible to teach dataview to correctly complete recurring tasks.

There are some other options for Tasks development that I'm toying with, that might make it easier to incorporate this calendar code in to Tasks.

Also, there is the dataview API which might allow this code to be incorporated in to the Tasks plugin without having to rewrite all the dataview code.

However, all of these are significant chunks of work, and I don't know if or when they would be done.
